watertap/core/util/misc.py

```diff
@@ -29,3 +30,16 @@ def is_constant_up_to_units(expr):
             continue
         return False
     return True
+
+
+def smooth_heaviside(x, k):
```

Comment:
How is this different then smooth_min, smooth_max in ideas? Looks pretty similar in operation, just configured to default out to 0 or the value you want?

Reply:
`Smooth_min `and `smooth_max `have also been implemented in the BSM2 translator blocks, but not all of the Expr_Ifs can be rewritten as `smooth_min `or `smooth_max`, so `smooth_heaviside `is being applied in these cases

Reply:
Makes sense, it might make sense to put a PR up on IDAES to add this function to their math file as I imagine this has broad utility and it would make sense to merge it out of WaterTAP to IDAES.

Reply:
Put up a PR in IDAES here: https://github.com/IDAES/idaes-pse/pull/1603